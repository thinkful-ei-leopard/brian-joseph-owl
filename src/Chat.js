import React from 'react';
import './Chat.css';
import ChatNonMessage from './ChatNonMessage';


function Chat(props) {

    // let chatId = props.participantId
    // let chatName = props.participants[chatId].name;
    // This doesnt work for some reason
    // chatName prints to the console, but no matter how we try to pass
    // it to ChatNonMessage, React always breaks by saying its undefined or an object
    // console.log(chatName);
    if(props.type !== 'message') {
        return <ChatNonMessage key={props.timestamp} name='Steve' type={props.type} />
    } else {
        let chatDate = new Date(props.timestamp);

        return (
            <div className="Chat-event">
                <img src="DOESNT WORK EITHER" alt="Not working" />
                <span><strong>{props.name}</strong></span>
                <span>{chatDate.toLocaleTimeString()}</span>
                <p className="Chat-Message">{props.message}</p>

            </div>
        );
    }
    
};

export default Chat;