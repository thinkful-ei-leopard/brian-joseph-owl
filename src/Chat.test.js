import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from './Chat';

describe('<Chat />', () => {
  // smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');

      ReactDOM.render(<Chat/>, div);

      ReactDOM.unmountComponentAtNode(div);
    });
  //snapshot test
    it('renders this UI as expected', () => {
      const tree = renderer.create(<Chat />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
