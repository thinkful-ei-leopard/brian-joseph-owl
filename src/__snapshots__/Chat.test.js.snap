// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders this UI as expected 1`] = `<div />`;
